### PR TITLE
Add dotnet builder

### PR DESCRIPTION
**Pull request by tomhobson**

### CODE REVIEW

**Comment by tomhobson** [comment]
Just thought I'd mention now -- I haven't removed `build-and-push-dlang`, it appeared in 2 batches so I thought I'd just replace dlang with dotnet in the first.

**Comment by abayer** [comment]
Please replace dlang in the place where you switched it out and instead add the new builder after dlang in both places.

**Comment by tomhobson** [comment]
@abayer That's done for you. would you mind explaining why it's done twice?

**Comment by abayer** [comment]
/ok-to-test

@tomhobson - building on PRs and building for releases. =) For various reasons, we're not able to do parallel builds of the images for releases, but we are for PRs, hence the duplication.

Could you squash your commits and make sure the commit message starts with something like `chore: Update dotnet builder`? Thanks.

File: builder-dotnet/Dockerfile [hunk 1]
```diff
@@ -2,9 +2,11 @@ FROM gcr.io/jenkinsxio/builder-base:0.0.73
 
 RUN rpm -Uvh https://packages.microsoft.com/config/rhel/7/packages-microsoft-prod.rpm \
     && yum update \
```
Comment (abayer) [inline builder-dotnet/Dockerfile]: So I think this needs a `-y` - the build failed with
```
Retrieving https://packages.microsoft.com/config/rhel/7/packages-microsoft-prod.rpm
Preparing...                          ########################################
Updating / installing...
packages-microsoft-prod-1.0-1.el7     ########################################
Loaded plugins: fastestmirror, ovl
Determining fastest mirrors
 * base: mirror.scalabledns.com
 * epel: d2lzkl7pfhq30w.cloudfront.net
 * extras: mirror.fileplanet.com
 * updates: mirror.scalabledns.com
Resolving Dependencies
--> Running transaction check
---> Package epel-release.noarch 0:7-11 will be updated
---> Package epel-release.noarch 0:7-12 will be an update
--> Finished Dependency Resolution

Dependencies Resolved

================================================================================
 Package                Arch             Version           Repository      Size
================================================================================
Updating:
 epel-release           noarch           7-12              epel            15 k

Transaction Summary
================================================================================
Upgrade  1 Package

Total download size: 15 k
Is this ok [y/d/N]: Exiting on user command
Your transaction was saved, rerun it with:
 yum load-transaction /tmp/yum_save_tx.2019-11-29.13-25.ZlJiRf.yumtx
```

**Comment by abayer** [comment]
@tomhobson It'll rerun after you push, no need to do retest then. =)

**Comment by tomhobson** [comment]
Oh that makes it easier

**Comment by abayer** [comment]
So once this passes and merges, I'll need to make an update to https://github.com/jenkins-x/jenkins-x-versions to include the dotnet builder there - I may not get to that today, but I will on Monday if nothing else.

**Comment by tomhobson** [comment]
Ok great 🙌 thanks very much

**Comment by abayer** [comment]
Version stream PR opened, and a followup PR here to make sure that anything reference `builder-dotnet` downstream of here gets properly update opened. =) So the builder should be usable in the same way as all the other builders by tomorrow.

**Pull request merged**